src/utils/aepp-wallet-communication/rpc/rpc-client.ts

```diff
@@ -0,0 +1,258 @@
+  }
+
+  constructor ({ info, id, name, icons, connection, handlers: [onMessage, onDisconnect] }: {
+    id: string
+    connection: Connection
+    handlers: [Function, Function]
+    info?: Partial<RpcClientInfo>
+    name?: RpcClientInfo['name']
+    icons?: RpcClientInfo['icons']
+  }) {
+    this.id = id
+    this.connection = connection
+    this.info = { ...info, name, icons }
+
+    this.callbacks = new Map()
+    // ['connected', 'current']
+    this.addressSubscription = []
+    this.accounts = {}
+
+    this.sendMessage = sendMessage(this.connection)
+
+    const handleMessage = (msg: Message, origin: string): void => {
+      if (msg.jsonrpc !== '2.0') {
+        throw new InvalidRpcMessageError(JSON.stringify(msg))
+      }
+      onMessage(msg, origin)
+    }
+
+    const disconnect = (connection: Connection): void => {
+      this.disconnect(true)
+      onDisconnect(connection, this)
+    }
+
+    this.connection.connect(handleMessage, disconnect)
+  }
+
+  /**
+   * Update info
+   * @param info Info to update (will be merged with current info object)
+   */
+  updateInfo (info: Partial<RpcClientInfo>): void {
+    this.info = { ...this.info, ...info }
+  }
+
+  isSubscribed (): boolean {
+    return (this.addressSubscription.length > 0) && isValidAccounts(this.accounts)
+  }
+
+  /**
+   * Check if aepp has access to account
+   * @param address Account address
+   * @returns is connected
+   */
+  hasAccessToAccount (address: string): boolean {
+    return (address.length !== 0) && (this.addresses.find(a => a === address) != null)
+  }
+
+  /**
+   * Check if is connected
+   * @return is connected
+   */
+  isConnected (): boolean {
+    return (this.connection.isConnected()) &&
+      (this?.info?.status === RPC_STATUS.CONNECTED || this?.info?.status === RPC_STATUS.NODE_BINDED)
+  }
+
+  /**
+   * Get selected account
+   * @param options Options
+   * @return current account
+   */
+  getCurrentAccount ({ onAccount }: { onAccount?: string } = {}): string {
+    return onAccount ?? Object.keys(this.accounts.current ?? {})[0]
+  }
+
+  /**
+   * Disconnect
+   */
+  disconnect (forceConnectionClose: boolean = false): void {
+    this.info.status = RPC_STATUS.DISCONNECTED
+    this.addressSubscription = []
+    this.accounts = {}
+    forceConnectionClose || this.connection.disconnect()
+  }
+
+  /**
+   * Update accounts and sent `update.address` notification to AEPP
+   * @param accounts Current and connected accounts
+   * @param options Options
+   */
+  setAccounts (accounts: Accounts,
+    { forceNotification }: { forceNotification?: boolean} = {}): void {
+    if (!isValidAccounts(accounts)) {
+      throw new TypeError('Invalid accounts object. Should be object like: `{ connected: {}, selected: {} }`')
+    }
+    this.accounts = accounts
+    if (!(forceNotification ?? false)) {
+      // Sent notification about account updates
+      this.sendMessage(message(METHODS.updateAddress, this.accounts), true)
+    }
+  }
```

Comment:
```suggestion
  setAccounts (accounts: Accounts, forceNotification?: boolean): void {
    if (!isValidAccounts(accounts)) {
      throw new TypeError('Invalid accounts object. Should be object like: `{ connected: {}, selected: {} }`')
    }
    this.accounts = accounts
    if (!(forceNotification ?? false)) {
      // Sent notification about account updates
      this.sendMessage(message(METHODS.updateAddress, this.accounts), true)
    }
  }
```
suggesting to make `forceNotification` an independent param.